tools/scripts/build/ensure-env.js

```diff
@@ -15,6 +15,39 @@ const globalConfigPath = path.resolve(__dirname, '../../../config');
 const { FREECODECAMP_NODE_ENV } = process.env;
 const { locale } = env;
 
+if (FREECODECAMP_NODE_ENV === 'production') {
+  const locationKeys = [
+    'homeLocation',
+    'apiLocation',
+    'forumLocation',
+    'newsLocation'
+  ];
+  const deploymentKeys = ['locale', 'deploymentEnv', 'environment'];
+  const searchKeys = ['algoliaAppId', 'algoliaAPIKey'];
+  const donationKeys = ['stripePublicKey', 'paypalClientId'];
+
+  const expectedVariables = locationKeys.concat(
+    deploymentKeys,
+    searchKeys,
+    donationKeys
+  );
+  const variables = Object.keys(env);
+  expectedVariables.sort();
+  variables.sort();
+  if (expectedVariables.length !== variables.length) {
+    throw Error(`Env. variable validation failed. Expected
+    ${variables} to match
+    ${expectedVariables}
+    `);
+  }
+
+  for (const key of variables) {
```

Comment:
```suggestion
  for (const key of expectedVariables) {
```
I'll make this change this while I rebase.